tiledb/sm/c_api/tiledb.cc

```diff
@@ -2896,37 +2896,6 @@ int32_t tiledb_deserialize_array_non_empty_domain_all_dimensions(
   return TILEDB_OK;
 }
 
-int32_t tiledb_serialize_array_max_buffer_sizes(
-    tiledb_ctx_t* ctx,
-    const tiledb_array_t* array,
-    const void* subarray,
-    tiledb_serialization_type_t serialize_type,
-    tiledb_buffer_t** buffer) {
-  // Sanity check
-  if (sanity_check(ctx, array) == TILEDB_ERR)
-    return TILEDB_ERR;
-
-  auto buf = tiledb_buffer_handle_t::make_handle(
-      ctx->resources().serialization_memory_tracker()->get_resource(
-          tiledb::sm::MemoryType::SERIALIZATION_BUFFER));
-
-  // Serialize
-  if (SAVE_ERROR_CATCH(
-          ctx,
-          tiledb::sm::serialization::max_buffer_sizes_serialize(
-              array->array_.get(),
-              subarray,
-              (tiledb::sm::SerializationType)serialize_type,
-              buf->buffer()))) {
-    tiledb_buffer_handle_t::break_handle(buf);
-    return TILEDB_ERR;
-  }
-
-  *buffer = buf;
-
-  return TILEDB_OK;
-}
```

Comment:
You can keep the API and make it throw if you don't want to wait for downstream changes.

Reply:
@teo-tsirpanis good idea! Let's do that if the downstream changes don't get traction in the next few days.

Reply:
Done.